propagators/propagator-aws-xray/src/AWSXRayPropagator.ts

```diff
@@ -66,48 +73,74 @@ export class AWSXRayPropagator implements TextMapPropagator {
     const timestamp = otTraceId.substring(0, TRACE_ID_FIRST_PART_LENGTH);
     const randomNumber = otTraceId.substring(TRACE_ID_FIRST_PART_LENGTH);
 
+    const xrayTraceId = `${TRACE_ID_VERSION}${TRACE_ID_DELIMITER}${timestamp}${TRACE_ID_DELIMITER}${randomNumber}`;
+
     const parentId = spanContext.spanId;
     const samplingFlag =
       (TraceFlags.SAMPLED & spanContext.traceFlags) === TraceFlags.SAMPLED
         ? IS_SAMPLED
         : NOT_SAMPLED;
     // TODO: Add OT trace state to the X-Ray trace header
 
-    const traceHeader = `Root=1-${timestamp}-${randomNumber};Parent=${parentId};Sampled=${samplingFlag}`;
+    let traceHeader =
+      `${TRACE_ID_KEY}` +
+      `${KV_DELIMITER}` +
+      `${xrayTraceId}` +
+      `${TRACE_HEADER_DELIMITER}` +
+      `${PARENT_ID_KEY}` +
+      `${KV_DELIMITER}` +
+      `${parentId}` +
+      `${TRACE_HEADER_DELIMITER}` +
+      `${SAMPLED_FLAG_KEY}` +
+      `${KV_DELIMITER}` +
+      `${samplingFlag}`;
+
+    const baggage = propagation.getBaggage(context);
+    const lineageV2Header = baggage?.getEntry(LINEAGE_KEY)?.value;
+
+    if (lineageV2Header) {
+      traceHeader +=
+        `${TRACE_HEADER_DELIMITER}` +
+        `${LINEAGE_KEY}` +
+        `${KV_DELIMITER}` +
+        `${lineageV2Header}`;
+    }
+
     setter.set(carrier, AWSXRAY_TRACE_ID_HEADER, traceHeader);
   }
 
   extract(context: Context, carrier: unknown, getter: TextMapGetter): Context {
-    const spanContext = this.getSpanContextFromHeader(carrier, getter);
-    if (!isSpanContextValid(spanContext)) return context;
-
-    return trace.setSpan(context, trace.wrapSpanContext(spanContext));
+    return this.getContextFromHeader(context, carrier, getter);
   }
 
   fields(): string[] {
     return [AWSXRAY_TRACE_ID_HEADER];
   }
 
-  private getSpanContextFromHeader(
```

Comment:
Logic overall lgtm.

`getContextFromHeader` is probably not an accurate method name anymore after these changes.
Since all the extract logic is now here, I think it might make sense to just move all this up into the `extract` method.